### ISSUE TITLE
Problems with same name images

### ISSUE THREAD

**Issue by UriHerrera** [Nitrux/znx#57]
There seems to be a problem if two deployed images have the same name even if they are on separate devices but are on the same computer.

For example, if `nx/dev` exists in `/dev/sda` and `/dev/nvmen0n1` booting any will cause that the graphical session never starts.

The problem is reproducible and can only be fixed by renaming one of the images.

**Comment by luis-lavaire** [comment]
@UriHerrera, could you please paste the output of `cat /proc/cmdline` for both images.

**Comment by UriHerrera** [comment]
They are the same image, so they use the same boot arguments. 

https://github.com/Nitrux/nitrux-iso-tool/blob/master/configs/grub.cfg

**Comment by luis-lavaire** [comment]
Oh, I can see why this happens. The images are based on Ubuntu or Debian. It's due to this parameter that the boot process fails: `iso-scan/filename=/STORE/nx/dev/IMAGE.0`. It doesn't specify the device on which the image should be found, and the initramfs picks-up the first one, regardless if that is the correct one.

**Comment by luis-lavaire** [comment]
The solution in theory:

Add another kernel parameter that specifies the UUID of the actual device. It just depends on whether `GRUB` has the means to provide such data.

**Comment by luis-lavaire** [comment]
Solved in d4cec4b.

The initramfs should process the value of `znx_dev_uuid`. It contains the UUID of the device from which the image booted.

For instance:

```shell
for k in $(cat /proc/cmdline); do
    case $k in
        ZNX_DEV_UUID=*)
            ZNX_DEV_UUID=${k#ZNX_DEV_UUID=}
        ;;
    esac
done

DEVICE=$(lsblk -no PATH,PARTUUID | grep $ZNX_DEV_UUID | cut -d ' ' -f 1)
```

**Comment by luis-lavaire** [comment]
This issue is not related to `znx` anymore. The boot scripts in `nitrux-iso-tool` are the ones that are wrong.

**Comment by UriHerrera** [comment]
Was the code merged into development? This commit 

> Solved in d4cec4b.

is from a non-existent branch.

**Comment by luis-lavaire** [comment]
Yes, it was merged.
